sklearn/cluster/_optics.py

```diff
@@ -348,7 +348,7 @@ def compute_optics_graph(X, *, min_samples, max_eps, metric, p, metric_params,
 
     Parameters
     ----------
-    X : array, shape (n_samples, n_features), or (n_samples, n_samples)  \
+    X : array of shape (n_samples, n_features), or (n_samples, n_samples)  \
```

Comment:
Same here (`array-like`).

Reply:
ok

Reply:
actually, should this be ndarray? first operation in the function calls X.shape

Reply:
Good catch, should be `ndarray`!

Reply:
Great, thanks, made that change now